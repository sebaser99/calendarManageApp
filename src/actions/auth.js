import Swal from "sweetalert2"
import { fetchConToken, fetchSinToken } from "../helpers/fetch"
import { types } from "../types/types"

export const startLogin = (email, password) => {
    return async (dispatch)=> {
        const resp = await fetchSinToken('auth', {email, password}, 'POST')
        const body = await resp.json()
        
        if(body.ok){
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime())

            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
        } else {
            Swal.fire('Error', body.msg, 'error')
        }
    }
}

export const startResgister = (name, email, password) => {
    return async (dispatch)=> {
        const resp = await fetchSinToken('auth/new', {email, name, password}, 'POST')
        const body = await resp.json()

        if(body.ok){
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime())
            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
        } else {
            return Swal.fire('Error', body.msg, 'error')
        }
    }
}

export const startChecking = () => {
    return async (dispatch)=> {
        const resp = await fetchConToken('auth/renew')
        const body = await resp.json()

        if(body.ok){
            localStorage.setItem('token', body.token)
            localStorage.setItem('token-init-date', new Date().getTime())
            dispatch(login({
                uid: body.uid,
                name: body.name
            }))
        } else {
                
                dispatch(checkingFinish())
            
        }
    }
}

export const checkingFinish = () => ({type: types.authCheckingFinish})


export const login = (user) => ({
    type: types.authLogin,
    payload: user
})

export const startLogout = () => {
    return (dispatch) => {
        localStorage.clear()
        dispatch(logout())
    }
}

export const logout = () => ({
    type: types.authLogout
})