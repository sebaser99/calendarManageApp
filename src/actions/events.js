import Swal from "sweetalert2";
import { fetchConToken } from "../helpers/fetch";
import { prepareEvent } from "../helpers/prepareEvent";
import { types } from "../types/types";

export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {
        const {name, uid} = getState().auth
        
        try{

            const res = await fetchConToken('events/new', event, 'POST')
            const body = await res.json();

            if(body.ok){
                event.id = body.event.id
                event.user = {
                    _id: uid,
                    name: name
                }

                dispatch(eventAddNew(event))
            }    

        } catch(error){
            console.log(error)
        }

    }
}

const eventAddNew = (event)=> ({
    type: types.eventsAddNew,
    payload: event
})

export const eventSetActive = (event)=> ({
    type: types.eventsSetActive,
    payload: event
})

export const clearActiveEvent = ()=> ({type: types.eventClearActive})

export const startUpdateEvent = (event) => {
    return async (dispatch) => {
        try {
            const res = await fetchConToken(`events/${event.id}`, event, 'PUT')
            const body = await res.json()

            if(body.ok){
                dispatch(updateEvent(event))
            } else {
                 Swal.fire('Error', body.msg, 'error')
            }  
        } catch(error){
         console.log(error)    
        }
    }
}

const updateEvent = (event) => ({
    type: types.eventUpdated,
    payload: event
})

export const startEventDelete = () => {
    
    return async (dispatch, getState) => {
        const {id} = getState().calendar.activeEvent
        try{
            const res = await fetchConToken(`events/${id}`, {}, 'DELETE')
            const body = await res.json();
            if(body.ok){
                dispatch(deletedEvent())
                
            } else {
                Swal.fire('Error', body.msg, 'error')
            }
            Swal.fire('Borrado ', 'Borrado Exitosamente', 'success')

        } catch(error){
            console.log(error)
            
        }
    }
          
}


const deletedEvent = () =>({type: types.eventDeleted})

export const slotNewEvent = (slotEvent) => ({
    type: types.eventSlotNew,
    payload: slotEvent
})

export const slotEventDelete = () => ({
    type: types.eventSlotDelete
})

export const eventsStartLoaded = () => {
    return async (dispatch) => {
        try{
            const res = await fetchConToken('events/')
            const body = await res.json()
            const eventos = prepareEvent(body.eventos) 
            dispatch(eventsLoaded(eventos))
         
        } catch(error) {
            console.log(error)
        }
    }
}

const eventsLoaded = (events) => ({
    type: types.eventLoaded,
    payload: events

})

export const eventsLogout = () =>(
        {
            type: types.eventLogout
            }
    )
        

