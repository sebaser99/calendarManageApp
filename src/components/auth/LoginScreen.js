import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { startLogin } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';
import '../../style/login.css';

export const LoginScreen = () => {
    const dispatch = useDispatch()
    const navigate = useNavigate()  

    const [formValue, handleChangeInput]= useForm({
        lEmail: "",
        lPassword: ""
    })
    const {lEmail, lPassword} = formValue

    const handleLogin = (e) => {
        e.preventDefault()
        dispatch(startLogin(lEmail, lPassword))
    }
    const handleNewAccount = ()=> {
        if(process.env.NODE_ENV !== 'test'){
            navigate('/auth/register')
        }
        
    }

  return (
        <div className="container login-container">
                   
            <div className="col-md-6 login-form-1">
                <h3>Ingreso</h3>
                <form onSubmit={handleLogin}>
                    <div className="form-group">
                        <input 
                            type="text"
                            className="form-control"
                            placeholder="Correo"
                            name="lEmail"
                            value={lEmail}
                            onChange={handleChangeInput}
                        />
                    </div>
                    <div className="form-group">
                        <input
                            type="password"
                            className="form-control"
                            placeholder="ContraseÃ±a"
                            name="lPassword"
                            value={lPassword}
                            onChange={handleChangeInput}
                        />
                    </div>
                    <div className="form-group">
                        <input 
                            type="submit"
                            className="btnSubmit"
                            value="Login" 
                        />
                    </div>
                </form>
                <div className='px-5 overline' onClick={handleNewAccount}>Create a new Account</div>
            </div>
        
                </div>          
     
  );
};
