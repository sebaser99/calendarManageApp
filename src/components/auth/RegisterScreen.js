import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import Swal from 'sweetalert2';
import { startResgister } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';
import '../../style/login.css';

export const RegisterScreen = () => {
    const dispatch = useDispatch()
    if(process.env.NODE_ENV !== 'test'){
        const navigate = useNavigate()
    }
    const [registerFormValues, handleInputChange] = useForm({
        rEmail: "sebaser99@hotmail.com", 
        rName: "Alex Rodriguez", 
        rPassword1: "1234567", 
        rPassword2: "1234567"
    })
    const {rEmail, rName, rPassword1, rPassword2} = registerFormValues

    const handleRegister = (e) => {
        e.preventDefault();
        if(rPassword1 !== rPassword2){
            return Swal.fire('Error', "Los passwords deben ser iguales", 'error')
        }
        dispatch(startResgister(rName, rEmail, rPassword1))

    }
    const handleBack = () => {
        if(process.env.NODE_ENV !== 'test'){
            navigate(-1)
        }
        
    }
  return (
    <div className="container login-container">
        <div className="col-md-8 login-form-2">
            <h3>Registro</h3>
            <form onSubmit={handleRegister}>
                <div className="form-group">
                    <input
                        type="text"
                        className="form-control"
                        placeholder="Nombre"
                        name="rName"
                        value={rName}
                        onChange={handleInputChange}
                    />
                </div>
                <div className="form-group">
                    <input
                        type="email"
                        className="form-control"
                        placeholder="Correo"
                        name="rEmail"
                        value={rEmail}
                        onChange={handleInputChange}
                    />
                </div>
                <div className="form-group">
                    <input
                        type="password"
                        className="form-control"
                        placeholder="Contraseña"
                        name="rPassword1"
                        value={rPassword1}
                        onChange={handleInputChange} 
                    />
                </div>

                <div className="form-group">
                    <input
                        type="password"
                        className="form-control"
                        placeholder="Repita la contraseña"
                        name="rPassword2"
                        value={rPassword2}
                        onChange={handleInputChange}  
                    />
                </div>

                <div className="form-group">
                    <input 
                        type="submit" 
                        className="btnSubmit" 
                        value="Crear cuenta" />
                </div>
            </form>
            <div className='px-5 overline blue' onClick={handleBack}> Volver</div>
        </div>
      </div>
  )
};
