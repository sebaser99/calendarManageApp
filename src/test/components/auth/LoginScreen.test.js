import React from 'react';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';

import configurStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { LoginScreen } from '../../../components/auth/LoginScreen';
import { startLogin } from '../../../actions/auth';


const middlewares = [thunk];
const mockStore = configurStore(middlewares);
const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn()
jest.mock('../../../actions/auth', ()=> ({
    startLogin: jest.fn()
}))

const wrapper = mount(
    <Provider store={store}>
        <LoginScreen/>
    </Provider>
)

describe('Vamos a probar el <LoginScreen/>', ()=> {
    
    test('debe hacer match con el snapshot', ()=> {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe llamar el dispatch de login', ()=> {
        wrapper.find('input[name="lEmail"]').simulate('change', {
            target: {
                name: 'lEmail',
                value: 'sebaser99@yahoo.com'
            }
        });
        wrapper.find('input[name="lPassword"]').simulate('change', {
            target: {
                name: 'lPassword',
                value: '1234567'
            }
        })

     
        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
            
        })

        // expect(startLogin).toHaveBeenCalledWith('sebaser99@yahoo.com', '1234567')
        expect(startLogin).toHaveBeenCalled()
    })

} )
