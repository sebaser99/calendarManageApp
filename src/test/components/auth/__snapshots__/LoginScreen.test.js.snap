// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vamos a probar el <LoginScreen/> debe hacer match con el snapshot 1`] = `
<div
  className="container login-container"
>
  <div
    className="col-md-6 login-form-1"
  >
    <h3>
      Ingreso
    </h3>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          name="lEmail"
          onChange={[Function]}
          placeholder="Correo"
          type="text"
          value=" "
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          name="lPassword"
          onChange={[Function]}
          placeholder="ContraseÃ±a"
          type="password"
          value=" "
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="btnSubmit"
          type="submit"
          value="Login"
        />
      </div>
    </form>
    <div
      className="px-5 overline"
      onClick={[Function]}
    >
      Create a new Account
    </div>
  </div>
</div>
`;
